internal/config/context.go

```diff
@@ -101,7 +101,7 @@ func CurrentContext() (endpoint Endpoint, user *User) {
 func GetCurrentContextInfo() (server string, username string, password string) {
 	endpoint, user := CurrentContext()
 	server = fmt.Sprintf("%s,%d", endpoint.Address, endpoint.Port)
-	if user != nil {
+	if user != nil && user.AuthenticationType == "basic" {
 		username = user.BasicAuth.Username
 		if user.AuthenticationType == "basic" {
```

Comment:
>	if user.AuthenticationType == "basic" { [](http://example.com/codeflow?start=1&length=40)

this line is redundant. 
It also implies that there could be a case where we still need a user name even when authenticationType is not "basic". 

What will the CurrentContext return for authentication types like AAD interactive that have a user name but no password?

Reply:
In main() this is only used if sqlcmd is invoked without any parameters and current-context is set in config file.
In that case, it is possible that the current-context may have a user associated with that context with AuthnticationType "Other".
`sqlcmd config add-user` generally creates user with 'Basic' auth-type as default but one could explicitly set this type as 'other'.